web/test/pages/profile.test.tsx

```diff
@@ -2319,10 +2320,53 @@ describe("profile", () => {
       expect(screen.queryByTestId("opp-alert")).not.toBeInTheDocument();
     });
 
+    it("does display date of formation question when legal structure is undefined", () => {
+      const userData = generateUserData({
+        profileData: {
+          ...emptyProfileData,
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+        },
+      });
+      renderPage({ userData });
+      expect(screen.getByTestId("question-dateOfFormation-alert-text")).toBeInTheDocument();
+      expect(screen.getByTestId("effective-date")).toBeInTheDocument();
+    });
+
+    it("does not display date of formation question when legal structure is defined and has a trade name", () => {
+      const userData = generateUserData({
+        profileData: {
+          ...emptyProfileData,
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+          legalStructureId: "general-partnership",
```

Comment:
we should use our randomized helpers here: `randomTradeNameBusiness` instead of only testing one case

Reply:
using randomTradeNameLegalStructure